agent/download.go

```diff
@@ -133,7 +133,7 @@ func (d Download) try() error {
 	}
 
 	// Now make the folder for our file
-	err = os.MkdirAll(targetDirectory, 0775)
+	err = os.MkdirAll(targetDirectory, 0777)
```

Comment:
As discussed, maybe we should put a brief comment on each of these, something like:

```suggestion
	err = os.MkdirAll(targetDirectory, 0777) // actual file mode reduced by system umask
```